feat(autocomplete): select profile to use for autocomplete

alexandrevilain opened the pull request:
## Context

Having only `google/gemini-2.5-flash-preview-05-20` is quite limited.
Added a way for users to select a profile for autocomplete so the user can choose the provider and the model he wants.

## Implementation

Added a new configuration item `autocompleteApiConfigId` that helps keep track of the selected profile to use for autocomplete.

## Screenshots

<img width="383" alt="image" src="https://github.com/user-attachments/assets/044022f1-f7ab-45d3-a06e-cc7f72572dc6" />

## How to Test

- Checkout this PR, run the extension.
- Go to experimental settings and select "Use experimental "autocomplete" feature"
- You should see the form to select the profile you want to use for autocomplete

## Get in Touch

<!-- We'd love to have a way to chat with you about your changes if necessary. If you're in the [Kilo Code Discord](https://kilocode.ai/discord), please share your handle here. -->

hassoncs commented:
Hey @alexandrevilain– Thanks for opening this PR! Yeah, I totally agree that the autocomplete should be configurable. I'll rebase this PR and get it merged asap!

Commit:
feat(autocomplete): Allow configuring autocomplete API provider

This commit introduces the ability to configure the API provider for the autocomplete feature.

Changes include:
- Added `autocompleteApiConfigId` to `globalSettingsSchema` and `ExtensionState`.
- Integrated `autocompleteApiConfigId` into `ClineProvider` and `webviewMessageHandler` for persistence and state management.
- Updated `AutocompleteProvider` to dynamically build the API handler based on the configured `autocompleteApiConfigId` using a new `getAutocompleteConfiguration` utility.
- Enhanced `AutocompleteStatusBar` to display model information and token validity, and updated its internationalization.
- Added new `AutocompletePromptSettings` component to the webview UI, allowing users to select an API configuration for autocomplete within the experimental settings.
- Updated various `kilocode.json` locale files to include new strings for autocomplete status bar and settings.

This allows users to select a specific API configuration for autocomplete, providing more flexibility and control over the feature's backend.

hassoncs reviewed (approved):
Merging! Thanks again, @alexandrevilain!! We really appreciate your effort!

 It should go out with the next release 🙏
